File: 백준/Silver/2839. 설탕 배달/설탕 배달.js
let input = require('fs')
.readFileSync('/dev/stdin')
.toString().split('\n');

let number = Number(input[0]);
let shareCount = 0;
let flag = false;

while(number>=0){
    if(number== 0 || number % 5 == 0) {
        shareCount += Math.floor(number/5);
        console.log(shareCount);
        flag = true;
        break;
    }
    number -= 3;
    shareCount += 1;
}
if(!flag){
    console.log(-1);
}

File: 백준/Silver/2839. 설탕 배달/설탕 배달.cs
using System;

class Program
{
    static void Main()
    {
        int number = int.Parse(Console.ReadLine());
        int shareCount = 0;
        bool flag = false;
        
        while (number >= 0)
        {
            if (number == 0 || number % 5 == 0)
            {
                shareCount += number/5;
                flag = true;
                Console.WriteLine(shareCount);
                break;
            }
            number -= 3;
            shareCount += 1;
        }
        if(!flag)
        {
            Console.WriteLine(-1);
        }
    }
}
